examples/todo-list/flutter-todo-list/lib/screens/todo.dart

```diff
@@ -0,0 +1,211 @@
+import 'package:flutter/material.dart';
+import 'package:todosupabase/constant.dart';
+import 'package:todosupabase/functions/auth.dart';
+import 'package:todosupabase/functions/crud.dart';
+
+class TodoPage extends StatefulWidget {
+  const TodoPage({Key? key}) : super(key: key);
+
+  @override
+  State<TodoPage> createState() => _TodoPageState();
+}
+
+class _TodoPageState extends State<TodoPage> {
+  String userName = "";
+  bool loading = true;
+  List<dynamic> todos = [];
+  @override
+  void initState() {
+    // TODO: implement initState
+    super.initState();
+    CrudSupabase.getUser(
+            email: AuthSupabase.client.auth.currentUser!.email.toString())
+        .then((value) {
+      setState(() {
+        userName = value[0]['name'];
+      });
+    });
+    CrudSupabase.getTodo().then((value) {
+      value.forEach((element) {
+        todos.add(element);
+      });
+    });
+    setState(() {
+      loading = false;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final TextEditingController _taskEditingController =
+        TextEditingController();
+    final TextEditingController _dueEditingController = TextEditingController();
+    return Scaffold(
+      floatingActionButton: FloatingActionButton(
+        backgroundColor: const Color(0xff33b27b),
+        onPressed: () {
+          displayTextInputDialog(context, () {
+            CrudSupabase.addTodo(
+              task: _taskEditingController.text,
+              due: _dueEditingController.text,
+            ).then((value) {
+              Navigator.pop(context);
+              setState(() {
+                loading = true;
+              });
+              CrudSupabase.getTodo().then((value) {
+                todos.clear();
+                setState(() {});
+                value.forEach((element) {
+                  todos.add(element);
+                });
+              });
+              setState(() {
+                loading = false;
+              });
+            });
+          }, _taskEditingController, _dueEditingController);
+        },
+        child: const Icon(Icons.add),
+      ),
+      appBar: AppBar(
+        backgroundColor: Colors.black,
+        automaticallyImplyLeading: false,
+        title: Text('Hey ${userName.split(" ")[0]}, add todo'),
+        actions: <Widget>[
+          IconButton(
+            icon: const Icon(Icons.exit_to_app),
+            onPressed: () {
+              AuthSupabase.logOut(context);
+            },
+          ),
+        ],
+      ),
+      body: loading
+          ? const Center(child: CircularProgressIndicator())
+          : SizedBox(
+              height: double.infinity,
+              width: double.infinity,
+              child: todos.isEmpty
+                  ? const Center(
+                      child: Text("No Task, add new task"),
+                    )
+                  : Column(
```

Comment:
Could we use a `ListView.build` maybe instead of using `Column` to display list of todos?

Reply:
sure I'll update the code